screens/DemoScreen.tsx

```diff
@@ -10,19 +10,53 @@ import {
 
 import { Text, View } from "../components/Themed";
 
+interface AdaptiveTextProps {
+  minFontSize: number;
+  maxTextLines: number;
+  children?: React.ReactNode;
+}
+
+const AdaptiveText = (props: AdaptiveTextProps) => {
+  const [hasHitBoundary, setHasHitBoundary] = useState(false);
+  
+  const [fontSize, setFontSize] = useState(props.minFontSize);
```

Comment:
To account for when the minimum font size would be _too large_, zero (`0`) may function as a better default; that way we can later compare whether or not we have too much text, and thus have shrunk below the minimum font size; we could then boost to the minimum font size, and enable the built-in `numOfLines` prop to trigger ellipsis.